### ISSUE TITLE
[timm][training] `eca_halonext26ts` fails to run

### ISSUE THREAD

**Issue by whitneywhtsang** [intel/intel-xpu-backend-for-triton#524]
`eca_halonext26ts` fails to run for all data types in training mode.

**Comment by ienkovich** [comment]
When I run the test, I get the error `RuntimeError: expected scalar type Long but found Int`. The stack points to the `aten.index_put_` operation called from inductor-generated code:
```
    buf34 = empty_strided((64, 80, 1, 144), (11520, 144, 144, 1), device='xpu', dtype=torch.float32)
    buf35 = reinterpret_tensor(buf34, (8, 640, 1, 12, 12), (92160, 144, 737280, 1, 12)); del buf34  # reuse
    # Source Nodes: [], Original ATen: [aten.clone, aten.unfold_backward]
    triton_poi_fused_clone_unfold_backward_11.run(buf35, buf32, 5120, 144, grid=grid(5120, 144), stream=stream0)
    del buf32
    buf36 = empty_strided((12, ), (1, ), device='xpu', dtype=torch.int32)
    # Source Nodes: [], Original ATen: [aten.unfold_backward]
    triton_poi_fused_unfold_backward_12.run(buf36, 12, grid=grid(12), stream=stream0)
    aten.index_put_(buf33, [None, None, None, reinterpret_tensor(buf36, (12, ), (1, ), 0)], buf35, True)
```
I cannot reproduce the error running this code, I get `index out of bounds` assert instead. It looks like the type check is missing in this case. This assertion can be easily reproduced though. Here is a test:
```
import torch
import intel_extension_for_pytorch

def test_index_put(device):
    print(f"Trying on {device}")
    lhs = torch.randn((10,), device=device, dtype=torch.float32)
    indices = torch.arange(10, device=device, dtype=torch.int32)
    rhs = torch.randn((10,), device=device, dtype=torch.float32)

    ref_res = lhs + rhs
    torch.ops.aten.index_put_(lhs, [indices], rhs, True)

    assert(torch.allclose(ref_res, lhs))

test_index_put('cpu')
test_index_put('xpu')
```
Which fails for me with:
```
Trying on cpu
Trying on xpu
/home/sdp/tongsu/frameworks.ai.pytorch.ipex-gpu/csrc/gpu/aten/operators/Indexing.h:736: operator(): global id: [0,0,0], local id: [0,0,0] Assertion `index >= -sizes[i] && index < sizes[i] && "index out of bounds"` failed
/home/sdp/tongsu/frameworks.ai.pytorch.ipex-gpu/csrc/gpu/aten/operators/Indexing.h:736: operator(): global id: [1,0,0], local id: [1,0,0] Assertion `index >= -sizes[i] && index < sizes[i] && "index out of bounds"` failed
/home/sdp/tongsu/frameworks.ai.pytorch.ipex-gpu/csrc/gpu/aten/operators/Indexing.h:736: operator(): global id: [2,0,0], local id: [2,0,0] Assertion `index >= -sizes[i] && index < sizes[i] && "index out of bounds"` failed
/home/sdp/tongsu/frameworks.ai.pytorch.ipex-gpu/csrc/gpu/aten/operators/Indexing.h:736: operator(): global id: [3,0,0], local id: [3,0,0] Assertion `index >= -sizes[i] && index < sizes[i] && "index out of bounds"` failed
/home/sdp/tongsu/frameworks.ai.pytorch.ipex-gpu/csrc/gpu/aten/operators/Indexing.h:736: operator(): global id: [4,0,0], local id: [4,0,0] Assertion `index >= -sizes[i] && index < sizes[i] && "index out of bounds"` failed
/home/sdp/tongsu/frameworks.ai.pytorch.ipex-gpu/csrc/gpu/aten/operators/Indexing.h:736: operator(): global id: [5,0,0], local id: [5,0,0] Assertion `index >= -sizes[i] && index < sizes[i] && "index out of bounds"` failed
/home/sdp/tongsu/frameworks.ai.pytorch.ipex-gpu/csrc/gpu/aten/operators/Indexing.h:736: operator(): global id: [6,0,0], local id: [6,0,0] Assertion `index >= -sizes[i] && index < sizes[i] && "index out of bounds"` failed
/home/sdp/tongsu/frameworks.ai.pytorch.ipex-gpu/csrc/gpu/aten/operators/Indexing.h:736: operator(): global id: [7,0,0], local id: [7,0,0] Assertion `index >= -sizes[i] && index < sizes[i] && "index out of bounds"` failed
/home/sdp/tongsu/frameworks.ai.pytorch.ipex-gpu/csrc/gpu/aten/operators/Indexing.h:736: operator(): global id: [8,0,0], local id: [8,0,0] Assertion `index >= -sizes[i] && index < sizes[i] && "index out of bounds"` failed
Aborted (core dumped)
```
The test passes if `torch.int64` is used for indices, so I guess the failure might be because its data is interpreted as int64 when `torch.int32` is used.

**Comment by vlad-penkin** [comment]
Fixed with in IPEX Triton 3.0 branch with commit id e6cd1dcc71e2a6d8fed995460fca7e97a130474d  on March 7th.

**Comment by vlad-penkin** [comment]
Run failure is no longer reproducible.

Env:
- pytorch is built from source, top of the main trunk, commit_id - `9a8ab778d34bd24c5caceb340837483decc4c311`
- triton xpu is built from source, top of the main trunk, commit_id - `fe93a00ffe438e9ba8c8392c0b051b1662c810de`
- benchmark is built from source, top of the main trunk, commit_id - `d54ca9f80ead108c8797441681e219becaf963d8`
- torchaudio is built from source, top of the main trunk,  commit_id - `1980f8af5bcd0bb2ce51965cf79d8d4c25dad8a0`
- torchvision  is built from source, top of the main trunk,  commit_id - `10239873229e527f8b7e7b3340c40ee38bb1cfc4`
- PyTorch Dependency Bundle 0.5.0
- Latest Rolling Driver
